dashboard/app/assets/javascripts/sendReport.js

```diff
@@ -43,34 +43,44 @@ var sendReport = function(report) {
 
   dashboard.clientState.trackProgress(report.result, report.lines, report.testResult, appOptions.level.scriptLevelId);
 
-  var thisAjax = jQuery.ajax({
-    type: 'POST',
-    url: report.callback,
-    contentType: 'application/x-www-form-urlencoded',
-    // Set a timeout of fifteen seconds so the user will get the fallback
-    // response even if the server is hung and unresponsive.
-    timeout: 15000,
-    data: queryString,
-    dataType: 'json',
-    beforeSend: function(xhr) {
-      xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content'));
-    },
-    success: function (response) {
-      if (thisAjax !== lastAjaxRequest) {
-        return;
+  //Post milestone iff the server tells us, or if we are on the last level and have passed
+  if (appOptions.postMilestone || appOptions.level.puzzle_number === appOptions.level.stage_total && report.pass) {
```

Comment:
Please parenthesize the && subexpression even though it's not strictly required.  (Of course && binds tigher than || but it's best not to have to even think about it.)